### PR TITLE
Vgp233/memogtt/moba camera

### PR DESCRIPTION
**Pull request by memogtt**
Added camera movement to work as MOBA cam, camera will be in the map limits checking mesh attributes of the map object.

### CODE REVIEW

File: WarSpecialists/Assets/Scripts/CameraMovement.cs [hunk 1]
```diff
@@ -5,16 +5,58 @@
 public class CameraMovement : MonoBehaviour
 {
     public GameObject target;
-    private float offset;
+    private float offset = 10.0f;   //for distance after pressing f to focus
```
Comment (bbi-ErickTavares) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs:8]: Serialize this variable that we can adjust this from the editor.

File: WarSpecialists/Assets/Scripts/CameraMovement.cs [hunk 2]
```diff
@@ -5,16 +5,58 @@
...
+    private float offsetBoundaries = 1.0f;
```
Comment (bbi-ErickTavares) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs:9]: Can you put a comment here? What this variable is doing?

File: WarSpecialists/Assets/Scripts/CameraMovement.cs [hunk 3]
```diff
@@ -5,16 +5,58 @@
...
 
-    void Awake()
+    //to variables to make it more readable
+    private float maxOffsetX, maxOffsetZ;
+
+    public GameObject map;
+    public float scrollSpeed = 20.0f;
```
Comment (bbi-ErickTavares) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs:15]: Make both variables private and serialize them.

File: WarSpecialists/Assets/Scripts/CameraMovement.cs [hunk 4]
```diff
@@ -5,16 +5,58 @@
...
+
+    void Start()
     {
         offset = transform.position.y;
+
+        //we got the map bounds size, and divide it by 2, because pivot is in middle
+        maxOffsetX = map.GetComponent<Renderer>().bounds.size.x / 2;
+        maxOffsetZ = map.GetComponent<Renderer>().bounds.size.z / 2;
+      //  offsetZ = maxOffsetZ
```
Comment (bbi-ErickTavares) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs:24]: Remove this comment.

**Review by bbi-ErickTavares — CHANGES_REQUESTED** [review]
Some few changes.
Is this script attached to the player?

**Review by erickbordallo — CHANGES_REQUESTED** [review]
I commented with my other account.

File: WarSpecialists/Assets/Scripts/CameraMovement.cs (hunk 1 above)
Comment (erickbordallo) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs]: Serialize this variable that we can adjust this from the editor.

File: WarSpecialists/Assets/Scripts/CameraMovement.cs (hunk 2 above)
Comment (erickbordallo) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs]: Can you put a comment here? What this variable is doing?

File: WarSpecialists/Assets/Scripts/CameraMovement.cs (hunk 3 above)
Comment (erickbordallo) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs]: Make both variables private and serialize them.

File: WarSpecialists/Assets/Scripts/CameraMovement.cs (hunk 4 above)
Comment (erickbordallo) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs]: Remove this comment.

**Review by erickbordallo — CHANGES_REQUESTED** [review]
Some small changes
Is this script attached to Camera? How it recognizes mouse position?

**Review by bbi-ErickTavares — APPROVED** [review]

**Comment by memogtt** [comment]
> Some small changes
> Is this script attached to Camera? How it recognizes mouse position?

the script is attached to the camera, it receives two references, one the map to stay inside its borders, and second the player so if you type f the camera will go back and focus quickly where the player is, it recognize mouse position by checking input mouse position in x and y

File: WarSpecialists/Assets/Scripts/CameraMovement.cs [hunk 5]
```diff
@@ -4,17 +4,68 @@
 
 public class CameraMovement : MonoBehaviour
 {
-    public GameObject target;
-    private float offset;
+    [Header("Player Settings")]
```
Comment (erickbordallo) [inline WarSpecialists/Assets/Scripts/CameraMovement.cs:7]: Great change!

**Review by erickbordallo — APPROVED** [review]
Awesome Job! You are free to Squash and Merge.

**Pull request merged**